src/backend/drm/compositor/mod.rs

```diff
@@ -1263,6 +1263,40 @@ where
         output: &Output,
         surface: DrmSurface,
         planes: Option<Planes>,
+        allocator: A,
+        framebuffer_exporter: F,
+        color_formats: &[DrmFourcc],
+        renderer_formats: HashSet<DrmFormat>,
+        cursor_size: Size<u32, BufferCoords>,
+        gbm: Option<GbmDevice<G>>,
+    ) -> FrameResult<Self, A, F> {
+        let output_mode_source = OutputModeSource::Auto(output.clone());
+        Self::new_with_output_mode_source(
+            output_mode_source,
+            surface,
+            planes,
+            allocator,
+            framebuffer_exporter,
+            color_formats,
+            renderer_formats,
+            cursor_size,
+            gbm,
+        )
+    }
+
+    /// Initialize a new [`DrmCompositor`] with custom output mode source.
+    ///
+    /// This method should only be used when trying to create a [`DrmCompositor`] that manually
+    /// updates its output mode through [`DrmCompositor::set_output_mode_source`]. If you want the
+    /// mode changes to be handled automatically, use [`DrmCompositor::new`] instead.
+    ///
+    /// See also: [`DrmCompositor::new`]
+    #[allow(clippy::too_many_arguments)]
+    #[instrument(skip(allocator, framebuffer_exporter))]
+    pub fn new_with_output_mode_source(
```

Comment:
This method could in theory be made private with a second method that just takes a size/scale/transform and handles this opaquely.

However I've found that for my compositor just implementing `From<MyOutput> for OutputModeSource` made this rather convenient and it reduces the number of parameters (which are already excessive), so I think just having this public should be perfectly fine (and maybe even better than the alternative).

Reply:
Maybe we can even simplify this into one `new`-function taking a `impl Into<OutputModeSource>` and implement that for `Output`?